daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dpath/DState.scala

```diff
@@ -215,45 +218,26 @@ case class DState(
   def booleanValue: Boolean = currentValue.getBoolean
 
   def longValue: Long = asLong(currentValue.getAnyRef)
+
   def intValue: Int = longValue.toInt
+
   def doubleValue: Double = asDouble(currentValue.getAnyRef)
 
   def integerValue: JBigInt = asBigInt(currentValue.getAnyRef)
+
   def decimalValue: JBigDecimal = asBigDecimal(currentValue.getAnyRef)
+
   def stringValue: String = currentValue.getString
 
   def isNilled: Boolean = currentElement.isNilled
 
-  private def isAnArray(): Boolean = {
-    if (!currentNode.isInstanceOf[DIArray]) {
-      Assert.invariant(errorOrWarn.isDefined)
-      if (currentNode.isInstanceOf[DIElement]) {
-        errorOrWarn.get.SDW(
-          WarnID.PathNotToArray,
-          "The specified path to element %s is not to an array. Suggest using fn:exists instead.",
-          currentElement.name,
-        )
-      } else {
-        errorOrWarn.get.SDW(
-          WarnID.PathNotToArray,
-          "The specified path is not to an array. Suggest using fn:exists instead.",
-        )
-      }
-      false
-    } else {
-      true
-    }
-  }
-
```

Comment:
So this is an 100% static check now? Nice! It might be worth adding a comment to the `arrayLength` and `finalArrayLength` functions that we must statically know current dpath result is an array, otherwise these might return some weird results or throw an exception. They are currently only used by FNCount so it should be safe, but if another function ever uses those we need to make sure they are doing the checkArgArray thing.

Reply:
I'm wondering if we can get away with just removing the warnID? The only place these warn IDs show up is with users disabling the warning. And if someone is disabling this specific warning then it probably means they have paths that don't resolve to an array. But this change now forces them to fix that since it's now an SDE, so also forcing them to remove references to that warning seems reasonable.